src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/GetDownloadResultUtility.cs

```diff
@@ -21,14 +24,19 @@ public static class GetDownloadResultUtility
            PackageIdentity identity,
            Uri uri,
            ISettings settings,
+           SourceCacheContext cacheContext,
            ILogger logger,
            CancellationToken token)
         {
+            // XXXXXXXXXXX This following assumption is false we constructed a Uri but we haven't attempted to use it yet
             // Uri is not null, so the package exists in the source
             // Now, check if it is in the global packages folder, before, getting the package stream
 
-            // TODO: This code should respect no_cache settings and not write or read packages from the global packages folder
-            var packageFromGlobalPackages = GlobalPackagesFolderUtility.GetPackage(identity, settings);
+            DownloadResourceResult packageFromGlobalPackages = null;
+
+            //TODO: This code should respect -NoCache option and not read packages from the global packages folder
```

Comment:
The TODO was already there - but it's related to NuGet/Home#1406 - I will update comment.